include/fastdds/dds/domain/DomainParticipantFactory.hpp

```diff
@@ -267,6 +269,42 @@ class DomainParticipantFactory
     RTPS_DllAPI ReturnCode_t set_qos(
             const DomainParticipantFactoryQos& qos);
 
+    /**
+     * @brief This operation returns the value of the DomainParticipant library settings.
+     *
+     * @param library_settings LibrarySettings reference where the settings are returned.
+     * @return RETCODE_OK
+     */
+    RTPS_DllAPI ReturnCode_t get_library_settings(
+            LibrarySettings& library_settings) const;
+
+    /**
+     * @brief This operation sets the library settings.
+     *
+     * Library settings must be set before enabling the DomainParticipants.
+     * Otherwise, failure of the setting operation is expected.
+     *
+     * @param library_settings LibrarySettings to be set.
+     * @return RETCODE_PRECONDITION_NOT_MET if any DomainParticipant is enabled.
```

Comment:
I have applied this comment only partially because I do not think we should dirty DDS documentation with RTPS references (moreover being implementation details that the user is not concerned about). The enable concept is included in the DDS specification and there is API to disable the automatic enabling of entities and API to enable created disabled entities.